$NetBSD: patch-bb,v 1.1.1.1 2007/03/12 16:08:30 briandealwis Exp $

This patch patches maillib.crm to use :fileprefix: for all :text_cache:
references.

--- /tmp/crm114-20061103-BlameDalkey.src/maillib.crm	2006-11-03 09:19:12.000000000 -0600
+++ maillib.crm	2007-02-23 16:03:14.000000000 -0600
@@ -439,7 +439,8 @@
 #       Reaver Cache routines
 #
 #           Assumptions= the var :text_cache: contains the name of
-#           the cache directory
+#           the cache directory; is still necessary to prefix with
+#           :fileprefix:
 #
 #      Assure that the text cache exists
 :reavercache_init:
@@ -450,15 +451,15 @@
                 #         and it's subdirectories.
                 #
                 isolate (:tmp:) //
-                syscall () (:tmp:) /ls :*:text_cache: 2>&1 /
+                syscall () (:tmp:) /ls :*:fileprefix::*:text_cache: 2>&1 /
                 match [:tmp:] <absent> /texts/
-                syscall () () /mkdir :*:text_cache: /
-                syscall () () /mkdir :*:text_cache:\/texts /
-                syscall () () /mkdir :*:text_cache:\/prob_good /
-                syscall () () /mkdir :*:text_cache:\/prob_spam /
-                syscall () () /mkdir :*:text_cache:\/known_good /
-                syscall () () /mkdir :*:text_cache:\/known_spam /
-                syscall () () /mkdir :*:text_cache:\/empty /
+                syscall () () /mkdir :*:fileprefix::*:text_cache: /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/texts /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/prob_good /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/prob_spam /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/known_good /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/known_spam /
+                syscall () () /mkdir :*:fileprefix::*:text_cache:\/empty /
         }
 }
 return
@@ -480,7 +481,7 @@
 #       It's unclear if the following increases security at all.
    isolate (:cacheid:) /:*:cacheid:_:*:msg_hash:/
    isolate (:long_cacheid:) /:*:text_cache:\/texts\/:*:cacheid:/
-   output [:*:long_cacheid:] /:*:text:/
+   output [:*:fileprefix::*:long_cacheid:] /:*:text:/
 }
 return
 
